### ISSUE TITLE
Please update install info, can't replicate a working version

### ISSUE THREAD

**Issue by DocMinus** [NREL/alfabet#15]
The installation info could be better. I didn't get it to run at all as described using pip install, because then upon import it complains that tensorflow is missing.
If I use the setup.py instead, then the scikit module complains about min. py3.8.
Finally, after all that (and a lot of wasted time), the tests_model.py fails due to uncompiled tensorflow model:

2023-01-04 15:48:50.354177: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-04 15:48:52.097245: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-04 15:48:52.744007: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 3849 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1080 Ti, pci bus id: 0000:82:00.0, compute capability: 6.1
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.

**Comment by GattiMh** [comment]
How did you complete the installation at the end?

**Comment by DocMinus** [comment]
> How did you complete the installation at the end?

I didn't, that's why I wrote here. But by now, I am sorry to say, I have put this on ice.

**Comment by GattiMh** [comment]
> > How did you complete the installation at the end?
> 
> I didn't, that's why I wrote here. But by now, I am sorry to say, I have put this on ice.

I couldn't install it on an M1 mac but on a linux no problem. Hope that helps

**Comment by totocotonio1** [comment]
it does not install

**Comment by DocMinus** [comment]
I managed a version which runs under Linux, unfortunately not with GPU (some library errors), but it works with CPU at least (tested on two entirely different machines) and works at least for regular py files (jupyter not included).
create a "environment.yml" file (text to paste in the end of this post) in the alfabet folder, then run 
```bash
conda env create -f environment.yml
```

if you want to test to see if it works at all without the jupyter stuff, here is a tiny script:
```Python
from alfabet import model

mm = model.predict(['CC', 'NCCO', "C1=CC=C(C=C1)[N+](=O)[O-]", 'CC(=O)O'])
mm.rename(columns={"bde_pred": "bde_pred(ML)", "bde": "bde(dft)"}, inplace=True)
mm.drop(columns=["bdfe_pred", "bdfe"], inplace=True)
print(mm)
```



in the file, paste this:

name: alfabetpy3.7
channels:
  - conda-forge
  - defaults
dependencies:
  - _libgcc_mutex=0.1
  - _openmp_mutex=4.5
  - boost=1.78.0
  - boost-cpp=1.78.0
  - brotli=1.1.0
  - brotli-bin=1.1.0
  - bzip2=1.0.8
  - ca-certificates=2024.2.2
  - cairo=1.16.0
  - certifi=2024.2.2
  - cycler=0.11.0
  - expat=2.5.0
  - font-ttf-dejavu-sans-mono=2.37
  - font-ttf-inconsolata=3.000
  - font-ttf-source-code-pro=2.038
  - font-ttf-ubuntu=0.83
  - fontconfig=2.14.2
  - fonts-conda-ecosystem=1
  - fonts-conda-forge=1
  - fonttools=4.38.0
  - freetype=2.12.1
  - gettext=0.21.1
  - greenlet=1.1.3
  - icu=72.1
  - importlib-metadata=4.11.4
  - jpeg=9e
  - kiwisolver=1.4.4
  - lcms2=2.14
  - ld_impl_linux-64=2.40
  - lerc=4.0.0
  - libblas=3.9.0
  - libbrotlicommon=1.1.0
  - libbrotlidec=1.1.0
  - libbrotlienc=1.1.0
  - libcblas=3.9.0
  - libdeflate=1.14
  - libexpat=2.5.0
  - libffi=3.4.2
  - libgcc-ng=13.2.0
  - libgfortran-ng=13.2.0
  - libgfortran5=13.2.0
  - libglib=2.78.4
  - libgomp=13.2.0
  - libiconv=1.17
  - liblapack=3.9.0
  - libnsl=2.0.1
  - libopenblas=0.3.26
  - libpng=1.6.42
  - libsqlite=3.45.1
  - libstdcxx-ng=13.2.0
  - libtiff=4.4.0
  - libuuid=2.38.1
  - libwebp-base=1.3.2
  - libxcb=1.13
  - libzlib=1.2.13
  - matplotlib-base=3.5.3
  - munkres=1.1.4
  - ncurses=6.4
  - numpy=1.21.6
  - openjpeg=2.5.0
  - openssl=3.2.1
  - packaging=23.2
  - pandas=1.3.5
  - pcre2=10.42
  - pillow=9.2.0
  - pip=24.0
  - pixman=0.43.2
  - pthread-stubs=0.4
  - pycairo=1.21.0
  - pyparsing=3.1.1
  - python=3.7.12
  - python-dateutil=2.8.2
  - python_abi=3.7
  - pytz=2024.1
  - rdkit=2022.09.1
  - readline=8.2
  - reportlab=3.5.68
  - setuptools=59.8.0
  - six=1.16.0
  - sqlalchemy=1.4.42
  - sqlite=3.45.1
  - tk=8.6.13
  - typing-extensions=4.7.1
  - typing_extensions=4.7.1
  - unicodedata2=14.0.0
  - wheel=0.42.0
  - xorg-kbproto=1.0.7
  - xorg-libice=1.1.1
  - xorg-libsm=1.2.4
  - xorg-libx11=1.8.4
  - xorg-libxau=1.0.11
  - xorg-libxdmcp=1.1.3
  - xorg-libxext=1.3.4
  - xorg-libxrender=0.9.10
  - xorg-renderproto=0.11.1
  - xorg-xextproto=7.3.0
  - xorg-xproto=7.0.31
  - xz=5.2.6
  - zipp=3.15.0
  - zlib=1.2.13
  - zstd=1.5.5
  - pip:
      - absl-py==2.1.0
      - astunparse==1.6.3
      - backcall==0.2.0
      - cachetools==5.3.2
      - charset-normalizer==3.3.2
      - debugpy==1.7.0
      - decorator==5.1.1
      - entrypoints==0.4
      - exceptiongroup==1.2.0
      - flatbuffers==23.5.26
      - gast==0.4.0
      - google-auth==2.28.1
      - google-auth-oauthlib==0.4.6
      - google-pasta==0.2.0
      - grpcio==1.62.0
      - h5py==3.8.0
      - idna==3.6
      - iniconfig==2.0.0
      - ipykernel==6.16.2
      - ipython==7.34.0
      - jedi==0.19.1
      - joblib==1.3.2
      - jupyter-client==7.4.9
      - jupyter-core==4.12.0
      - keras==2.11.0
      - libclang==16.0.6
      - markdown==3.4.4
      - markupsafe==2.1.5
      - matplotlib-inline==0.1.6
      - nest-asyncio==1.6.0
      - networkx==2.6.3
      - nfp==0.3.12
      - oauthlib==3.2.2
      - opt-einsum==3.3.0
      - parso==0.8.3
      - pexpect==4.9.0
      - pickleshare==0.7.5
      - platformdirs==4.0.0
      - pluggy==1.2.0
      - pooch==1.8.1
      - prompt-toolkit==3.0.43
      - protobuf==3.19.6
      - psutil==5.9.8
      - ptyprocess==0.7.0
      - pyasn1==0.5.1
      - pyasn1-modules==0.3.0
      - pygments==2.17.2
      - pytest==7.4.4
      - pyzmq==25.1.2
      - requests==2.31.0
      - requests-oauthlib==1.3.1
      - rsa==4.9
      - scikit-learn==0.24.2
      - scipy==1.7.3
      - tensorboard==2.11.2
      - tensorboard-data-server==0.6.1
      - tensorboard-plugin-wit==1.8.1
      - tensorflow==2.11.0
      - tensorflow-estimator==2.11.0
      - tensorflow-io-gcs-filesystem==0.34.0
      - termcolor==2.3.0
      - threadpoolctl==3.1.0
      - tomli==2.0.1
      - tornado==6.2
      - tqdm==4.66.2
      - traitlets==5.9.0
      - urllib3==2.0.7
      - wcwidth==0.2.13
      - werkzeug==2.2.3
      - wrapt==1.16.0

**Comment by zacbcaz** [comment]
For folks still waiting for a working version, it appears that the original authors have created some new/updated models while neglecting this version (including their web-based version).  The paper providing updates and a link to a new repository is below.  While not as easy as a pip install, I was able to get a working version going by downloading their code/models directly, following their example as provided.

https://doi.org/10.1039/D3DD00169E

https://github.com/patonlab/BDE-db2

**Comment by DocMinus** [comment]
> For folks still waiting for a working version, it appears that the original authors have created some new/updated models while neglecting this version (including their web-based version). The paper providing updates and a link to a new repository is below. While not as easy as a pip install, I was able to get a working version going by downloading their code/models directly, following their example as provided.
> 
> https://doi.org/10.1039/D3DD00169E
> 
> https://github.com/patonlab/BDE-db2

Thanks!
